dask/array/fft.py

```diff
@@ -28,31 +30,37 @@ def _fft_out_chunks(a, n, axis):
     return chunks
 
 
-def _rfft_out_chunks(a, n, axis):
+def _rfft_out_chunks(a, n, axis, rtype):
     if n is None:
         n = a.chunks[axis][0]
     chunks = list(a.chunks)
-    chunks[axis] = (n // 2 + 1,)
+    if issubclass(np.dtype(rtype).type, numbers.Real):
+        chunks[axis] = (n,)
+    else:
+        chunks[axis] = (n // 2 + 1,)
```

Comment:
Have added a `ValueError` for those specific SciPy functions and added tests that cover it. Also have reverted all of the special `dtype` handling code.